src/types.ts

```diff
@@ -0,0 +1,87 @@
+class SchemaNotFound extends Error {
+  constructor(uri) {
+    super();
+    const error = new Error("Schema not found: " + uri);
+    error.name = "SchemaNotFound";
+    return error;
+  }
+}
+
+
+/**
+ * @class BaseClient.Types
+ *
+ * - Manages and validates types of remoteStorage objects, using JSON-LD and
+ *   JSON Schema
+ * - Adds schema declaration/validation methods to BaseClient instances.
+ **/
+class Types {
+  SchemaNotFound = SchemaNotFound;
+
+  /**
+   * <alias> -> <uri>
+   */
+  uris: { [key: string]: string } = {};
+
+  /**
+   * Contains schema objects of all types known to the BaseClient instance
+   *
+   * <uri> -> <schema>
+   */
+  schemas: { [key: string]: string } = {};
+
+  /**
+   * <uri> -> <alias>
+   */
+  aliases: { [key: string]: string } = {};
+
+  /**
+   * Called via public function BaseClient.declareType()
+   *
+   * @private (though not really)
```

Comment:
The private in this JSDoc comment refers to the fact that it's not a public API for library consumers, not to being private within the library itself.

Reply:
Yes. The comment was probably not ideal to make my intentions clear...